core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/service/GitVersionControlLocalFileStorageService.java

```diff
@@ -0,0 +1,238 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+/**
+ * Git-based implementation of the VersionControlFileStorageService, using local file storage.
+ */
+public class GitVersionControlLocalFileStorageService {
+
+  /**
+   * Writes content from the InputStream to a file at the given path within the repository.
+   *
+   * @param repoPath The repository base path.
+   * @param relativeFilePath The relative path within the repository to write the file.
+   * @param inputStream The InputStream from which to read the content.
+   * @throws IOException If an I/O error occurs.
+   */
+  public static void writeFile(String repoPath, String relativeFilePath, InputStream inputStream) throws IOException {
+    Path filePath = Paths.get(repoPath, relativeFilePath);
+    Files.createDirectories(filePath.getParent());
+    Files.copy(inputStream, filePath, StandardCopyOption.REPLACE_EXISTING);
+  }
+
+  /**
+   * Deletes a file at the given path within the repository.
+   *
+   * @param repoPath The repository base path.
+   * @param relativeFilePath The relative path within the repository of the file to delete.
+   * @throws IOException If an I/O error occurs.
+   */
+  public static void deleteFile(String repoPath, String relativeFilePath) throws IOException {
+    Path filePath = Paths.get(repoPath, relativeFilePath);
+    Files.deleteIfExists(filePath);
+  }
+
+  /**
+   * Deletes the entire directory specified by the repository path.
+   *
+   * @param repoPath The path of the directory to delete.
+   * @throws IOException If an I/O error occurs.
+   */
+  public static void deleteDirectory(String repoPath) throws IOException {
+    Path directoryPath = Paths.get(repoPath);
+    Files.walk(directoryPath)
+        .sorted(Comparator.reverseOrder())
+        .map(Path::toFile)
+        .forEach(File::delete);
+  }
+
+  /**
+   * Initializes a new repository for version control at the specified path.
+   *
+   * @param baseRepoPath Path to initialize the repository at.
+   * @throws IOException If an I/O error occurs.
+   * @throws InterruptedException If the operation is interrupted.
+   */
+  public static void initRepo(String baseRepoPath) throws IOException, InterruptedException {
+    GitSystemCall.initRepo(baseRepoPath);
+  }
+
+  /**
+   * Creates a new version in the repository with the given version name.
+   *
+   * @param baseRepoPath The repository path.
+   * @param versionName The name or message associated with the version.
+   * @return The commit hash of the created version.
+   * @throws IOException If an I/O error occurs.
+   * @throws InterruptedException If the operation is interrupted.
+   */
+  public static String createVersion(String baseRepoPath, String versionName) throws IOException, InterruptedException {
+    return GitSystemCall.addAndCommit(baseRepoPath, versionName);
+  }
```

Comment:
first, the createVersion name is confusing. please consider change it to `bumpVersion` or `incrementVersion`. 

second, the current design has createVersion and other functions separated. The user may write code like this which forgets to invoke createVersion() after some changes:

```
gvclfs.addFile(file1)
gvclfs.addFile(file2)
gvclfs.createVersion()
gvclfs.addFile(file3)
// and then user forgets to call createVersion()
```

Can we make createVersion() inside each operation function so that user does not need to explicitly call it? E.g.,
```
gvclfs.addFile(file1) // version number is automatically bumped.
gvclfs.addFile(file2) // version number is automatically bumped.
```

Reply:
Done. The withCreatVersion style is good. I implemented one.